datafusion/sqllogictest/test_files/explain.slt

```diff
@@ -362,3 +362,18 @@ GlobalLimitExec: skip=0, fetch=10, statistics=[Rows=Exact(8), Bytes=Absent, [(Co
 
 statement ok
 set datafusion.execution.collect_statistics = false;
+
+# Explain ArrayFuncions
+
+statement ok
+set datafusion.explain.physical_plan_only = false
+
+query TT
+explain select make_array(make_array(1, 2, 3), make_array(4, 5, 6));
```

Comment:
Given there is a special case for `[]` can we also please add a test for that as well?